plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeConversions.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.orc;
+
+import io.trino.orc.metadata.OrcType.OrcTypeKind;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.IntArrayBlock;
+import io.trino.spi.type.Type;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import static io.trino.orc.metadata.OrcType.OrcTypeKind.BYTE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+
+public final class OrcTypeConversions
+{
+    private OrcTypeConversions() {}
+
+    public static Function<Block, Block> getTypeConversion(OrcTypeKind orcType, Type trinoType)
+    {
+        if (orcType == BYTE) {
+            if (trinoType == INTEGER) {
+                return OrcTypeConversions::tinyintToInteger;
+            }
+        }
+        return block -> block;
+    }
+
+    private static Block tinyintToInteger(Block block)
+    {
+        boolean[] isNull = new boolean[block.getPositionCount()];
+        int[] values = new int[block.getPositionCount()];
+        for (int i = 0; i < block.getPositionCount(); i++) {
+            if (block.isNull(i)) {
+                isNull[i] = true;
+            }
+            else {
+                values[i] = block.getByte(i, 0);
+            }
```

Comment:
I suspect this approach is going to be slower because of the additional null check. You can try adding a JMH benchmark like `BenchmarkOrcDecimalReader` and compare the performance of these approaches.
We're also not preserving the `mayHaveNull` of the original block, that should definitely be done.
We did these type of adaptations in column readers rather than the page source in the parquet reader as well.

Reply:
The primitive types of orc and parquet are different so the required adaptations won't be the same (e.g. there is no byte/short in parquet only int32/int64).
We could create a function in ByteColumnReader to avoid repeating code like below
```
if (type == TINYINT) {
      return new ByteArrayBlock(nextBatchSize, Optional.empty(), values);
}
 if (type == INTEGER) {
      return new IntArrayBlock(nextBatchSize, Optional.empty(), convertToIntArray(values));
}
throw new VerifyError("Unsupported type " + type);
```
These types of column adaptations which are done in the reader are simple ones meant to resolve minor differences in file format type and trino type (e.g. differing byte count of numbers, difference in precision of decimals, timestamps etc.). These should make sense for any connector. For anything complex like reading a number in the file format as a varchar trino type, which a connector may or may not want to do, can be done in the connector.